src/NUnitFramework/framework/Extensions.cs

```diff
@@ -39,7 +39,7 @@ public static bool HasAttribute<T>(this Type type, bool inherit)
 
         public static T[] GetAttributes<T>(this Type type, bool inherit) where T : class
         {
-            return ((ICustomAttributeProvider)type.GetTypeInfo()).GetAttributes<T>(inherit);
+            return ((ICustomAttributeProvider)type).GetAttributes<T>(inherit);
```

Comment:
Are these casts still required after removing GetTypeInfo?

Reply:
still required since they force the path tp go into `this ICustomAttributeProvider attributeProvider` path

TBH the whole reflection codebase need some re-work. but i would rather attempt that in another pr